mlflow/gateway/utils.py

```diff
@@ -98,6 +99,18 @@ def assemble_uri_path(paths: List[str]) -> str:
     return "/" + posixpath.join(*stripped_paths)
 
 
+def assemble_url(base_url: str, route: str) -> str:
```

Comment:
```suggestion
def join_url(base_url: str, route: str) -> str:
```

`join` might be simpler and easier to infer what this functions does.